api/v1beta1/csiaddonsjob_types.go

```diff
@@ -0,0 +1,117 @@
+/*
+Copyright 2021 The Kubernetes-CSI-Addons Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// TargetSpec defines the targets on which the operation can be
+// performed.
+type TargetSpec struct {
+	// PVC represents PersistentVolumeClaim name.
+	PVC string `json:"pvc,omitempty"`
+
+	// PVC represents PersistentVolume name.
+	PV string `json:"pv,omitempty"`
+}
+
+// ReclaimSpaceSpec defines the required parameters for ReclaimSpace
+// operation.
+type ReclaimSpaceSpec struct {
+	// Target represents targets on which the operation can be
+	// performed.
+	// +kubebuilder:validation:Required
+	Target TargetSpec `json:"target"`
+
+	// Parameters is a key-value map with driver configurations for
+	// performing the operation.
+	// +kubebuilder:validation:Optional
+	Parameters map[string]string `json:"parameters,omitempty"`
+}
+
+// OperationSpec defines the desired operation of CSIAdoonsJobSpec.
+type OperationSpec struct {
+	// ReclaimSpace repesents ReclaimSpace operation.
+	ReclaimSpace ReclaimSpaceSpec `json:"reclaimSpace,omitempty"`
+}
+
+// OperationState defines the state of the operation.
+type OperationState string
```

Comment:
Currently, this says operation "state" which definitionally means that it shows the ongoing status of a thing, however, the only values below are final result states. 

Should this be left as it is to support a "Processing" state or some other thing in the future?

Or would it be better to change this to `OperationResult` in order to indicate it should only represent the final result?